### ISSUE TITLE
`python.analysis.completeFunctionParens` adds parens on base class names

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#4273]
## Environment
-   Pylance version: 2023.4.30

## Code Snippet

```python
class BaseClass:
    pass
```

## Repro Steps

0. Ensure `python.analysis.completeFunctionParens` is set to `true`.
1. Create empty class as above.
2. Start creating derived class `class Foo(Bas` and press tab when `BaseClass` is suggested as the completion

## Expected behavior

No parens added after `BaseClass`:

```python
class Foo(BaseClass)
```
## Actual behavior
Parens are added, which results in an error: `Expected type expression but received "BaseClass"`
```python
class Foo(BaseClass())
```

**Comment by rchiodo** [comment]
Seems like a bad case. I think we should fix this.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.7.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023741-26-July-2023-PreRelease)
